Interaction Script

ZacOlsen opened the pull request:
If a character is close to an interactable object, a Sprite appears above the object. If the player pushes the interact button (currently arrow up) while the Sprite is being displayed, text (that describes how to interact with the object) appears in its place. There is also a bool called conversation that, if true, activates the objects conversation script (almost: I do not have the conversation script so I do not know how to call it, but I wrote an Invoke call for when there is one).

ZacOlsen committed:
dis change

changed

ZacOlsen committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DisBMyBranch

Conflicts:
	TutorialFile

ZacOlsen committed:
Interaction Script

What this does-

If a character is close to an interactable object, a Sprite appears
above the object. If the player pushes the interact button (currently
arrow up) while the Sprite is being displayed, text (that describes how
to interact with the object) appears in its place. There is also a bool
called conversation that, if true, activates the objects conversation
script (almost: I do not have the script so I do not know how to call
it, but i wrote an Invoke call for when there is one).

ZacOlsen committed:
something

idk

ashleycrouch commented:
Some things that this still needs to be able to do before it can be merged:
*needs to be able to cycle through text with the push of a button, and if there isn't anymore text, it closes (turns off the panel and everything) when you push a button
*have the text not go past the borders of the box

ZacOlsen committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DisBMyBranch

Conflicts:
	LurkTestBuild/Assets/Scripts/NPCs/Enemy.cs.meta
	LurkTestBuild/ProjectSettings/ProjectSettings.asset
	LurkTestBuild/ProjectSettings/ProjectVersion.txt
	TutorialFile

ZacOlsen committed:
commiting changes

ZacOlsen committed:
update from master

Veloxi committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DisBMyBranch

Veloxi committed:
Some Minor Changes, moved to new Scene

I made some minor changes, feel free to revert if you want, but please
read the comments I added first!

Veloxi commented:
Hey! I made some minor changes to your branch. I moved everything into your own testing scene, and i altered the script slightly according to my comments in the script. If you really want to you can revert but please read my comments first!
@ZacOlsen  @ashleycrouch

ZacOlsen committed:
Merge remote-tracking branch 'refs/remotes/origin/master' into DisBMyBranch

ZacOlsen committed:
update from master

ZacOlsen committed:
update interaction script

icon is now trigger when a player enters the radius of a collider

keys pushing has been removed for the player class to deal with, instead
of the StartKey and CycleKey methods will now be called to start the
interaction and cycle through it

canvas is no longer disabled, the display object holds the text and
panel children which is made into a child of canvas, this will be
activated and unactivated

instanceIcon has been replaced with icon

player array has been removed because of the collider

ZacOlsen committed:
Merge remote-tracking branch 'origin/DisBMyBranch' into DisBMyBranch

ZacOlsen committed:
added the prefab I used for the interaction script
